### PR TITLE
[4.0] [WIP] Use namespaced classes everywhere

### PR DESCRIPTION
**Pull request by C-Lodder**
Pull Request for Issue #24108

### Summary of Changes

I'm aware that there are still instances where the `J` prefixed class only resides in comment blocks and the `use` case is still added. My regex stills aren't top notch so I'll leave it up the the powers that be to decide whether or not they're fussed about this. If it's a problem I can close this and anyone willing to improve the regex can take over from me.

- JText
- JPath
- JFile
- JFolder
- JStream
- JFilesystemHelper
- JObject
- JImageFilter
- JImage
- JFeedPerson
- JFeedParser
- JFeedEntry
- JFeed
- JInputFiles
- JInputCli

### Testing Instructions

Browse through the frontend/backend and check for any errors

### CODE REVIEW

**Comment by brianteeman** [comment]
cool stuff

**Comment by brianteeman** [comment]
When the change is only in the docblock does it still need to have the use namespace statement?

**Comment by C-Lodder** [comment]
No. My regex initially dealt with comment blocks but then only detected classes with a leading backslash. Couldnt figure it out so just stuck with this

File: administrator/components/com_modules/Model/ModuleModel.php [hunk 1]
```diff
@@ -718,9 +719,9 @@ public function getItem($pk = null)
 				}
 			}
 
-			// Convert to the \JObject before adding other data.
+			// Convert to the CMSObject before adding other data.
 			$properties        = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (SharkyKZ) [inline administrator/components/com_modules/Model/ModuleModel.php]: This is going to error. It should be `CMSObject::class`, without quotes.

Comment (C-Lodder) [reply administrator/components/com_modules/Model/ModuleModel.php]: Not sure how the JObect would work in quotes but not CMSObject, seeing as it's just a mapped name

Comment (SharkyKZ) [reply administrator/components/com_modules/Model/ModuleModel.php]: It's because `CMSObject` is namespaced. You have to use fully qualified class name here. Since class is already imported, you can use `class` keyword for this.

Comment (C-Lodder) [reply administrator/components/com_modules/Model/ModuleModel.php]: Fair enough, I've updated it as requested. Thanks

File: administrator/components/com_menus/View/Items/HtmlView.php [hunk 2]
```diff
@@ -12,12 +12,13 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\Factory;
-use Joomla\CMS\Helper\ContentHelper;
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Helper\ContentHelper;
```
Comment (SharkyKZ) [inline administrator/components/com_menus/View/Items/HtmlView.php:16]: Revert this.

File: administrator/components/com_menus/View/Menu/XmlView.php [hunk 3]
```diff
@@ -14,8 +14,9 @@
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\Log\Log;
-use Joomla\CMS\Menu\MenuHelper;
 use Joomla\CMS\MVC\View\HtmlView as BaseHtmlView;
+use Joomla\CMS\Menu\MenuHelper;
```
Comment (SharkyKZ) [inline administrator/components/com_menus/View/Menu/XmlView.php:18]: Revert this.

File: components/com_contact/View/Contact/HtmlView.php [hunk 4]
```diff
@@ -13,10 +13,11 @@
 
 use Joomla\CMS\Categories\Categories;
 use Joomla\CMS\Factory;
-use Joomla\CMS\Helper\TagsHelper;
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Helper\TagsHelper;
```
Comment (SharkyKZ) [inline components/com_contact/View/Contact/HtmlView.php:17]: Revert this.

File: components/com_contact/View/Featured/HtmlView.php [hunk 5]
```diff
@@ -14,8 +14,9 @@
 use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Mail\MailHelper;
 use Joomla\CMS\MVC\View\HtmlView as BaseHtmlView;
+use Joomla\CMS\Mail\MailHelper;
```
Comment (SharkyKZ) [inline components/com_contact/View/Featured/HtmlView.php:18]: Revert this.

Comment (C-Lodder) [reply components/com_contact/View/Featured/HtmlView.php:18]: @SharkyKZ Hold off on PHPCS reviews for now until we know what the decision is

File: libraries/cms/html/behavior.php [hunk 6]
```diff
@@ -646,7 +646,7 @@ protected static function calendartranslation()
 		$jsscript = 1;
 
 		// To keep the code simple here, run strings through Text::_() using array_map()
-		$callback = array('JText', '_');
+		$callback = array('Text', '_');
```
Comment (SharkyKZ) [inline libraries/cms/html/behavior.php]:
```suggestion
		$callback = array(Text::class, '_');
```

File: components/com_contact/View/Featured/HtmlView.php (hunk 5 above)
Comment (SharkyKZ) [reply components/com_contact/View/Featured/HtmlView.php:18]: Well, there's a ton of issues. Not just CS. Like the ones mentioned before or this https://github.com/joomla/joomla-cms/pull/24174/files#diff-6bdca1b61c6581c3aa2a75ad80529353R16

Comment (C-Lodder) [reply components/com_contact/View/Featured/HtmlView.php:18]: Ahh I fixed some of those already but didn't see that one. Will soft through later on this evening

Comment (SharkyKZ) [reply components/com_contact/View/Featured/HtmlView.php:18]: By the way, are you suggesting that this build module be added to Joomla? Or is it here temporarily and would be removed before merging?

Comment (C-Lodder) [reply components/com_contact/View/Featured/HtmlView.php:18]: I can be added if others wish to help out and then removed once all the namespacing is done, or can be removed from the PR once before merging. I'm not fussed

File: administrator/components/com_languages/Model/LanguageModel.php [hunk 7]
```diff
@@ -124,7 +125,7 @@ public function getItem($langId = null)
 		}
 
 		$properties = $table->getProperties(1);
-		$value      = ArrayHelper::toObject($properties, 'JObject');
+		$value      = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (SharkyKZ) [inline administrator/components/com_languages/Model/LanguageModel.php]:
```suggestion
		$value      = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_menus/Model/MenuModel.php [hunk 8]
```diff
@@ -138,7 +139,7 @@ public function &getItem($itemId = null)
 		}
 
 		$properties = $table->getProperties(1);
-		$value      = ArrayHelper::toObject($properties, 'JObject');
+		$value      = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (SharkyKZ) [inline administrator/components/com_menus/Model/MenuModel.php]:
```suggestion
		$value      = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_plugins/Model/PluginModel.php [hunk 9]
```diff
@@ -175,9 +176,9 @@ public function getItem($pk = null)
 				return false;
 			}
 
-			// Convert to the \JObject before adding other data.
+			// Convert to the CMSObject before adding other data.
 			$properties = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (SharkyKZ) [inline administrator/components/com_plugins/Model/PluginModel.php]:
```suggestion
			$this->_cache[$pk] = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_templates/Model/StyleModel.php [hunk 10]
```diff
@@ -363,9 +364,9 @@ public function getItem($pk = null)
 				return false;
 			}
 
-			// Convert to the \JObject before adding other data.
+			// Convert to the CMSObject before adding other data.
 			$properties        = $table->getProperties(1);
-			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'JObject');
+			$this->_cache[$pk] = ArrayHelper::toObject($properties, 'CMSObject');
```
Comment (SharkyKZ) [inline administrator/components/com_templates/Model/StyleModel.php]:
```suggestion
			$this->_cache[$pk] = ArrayHelper::toObject($properties, CMSObject::class);
```

File: administrator/components/com_users/Model/GroupModel.php [hunk 11]
```diff
@@ -130,7 +131,7 @@ protected function loadFormData()
 	 */
 	protected function preprocessForm(\JForm $form, $data, $group = '')
 	{
-		$obj = is_array($data) ? ArrayHelper::toObject($data, 'JObject') : $data;
+		$obj = is_array($data) ? ArrayHelper::toObject($data, 'CMSObject') : $data;
```
Comment (SharkyKZ) [inline administrator/components/com_users/Model/GroupModel.php]:
```suggestion
		$obj = is_array($data) ? ArrayHelper::toObject($data, CMSObject::class) : $data;
```

File: libraries/src/Language/Language.php [hunk 12]
```diff
@@ -1006,7 +1007,7 @@ protected function getCallerInfo()
 			$class = @ $step['class'];
 
 			// We're looking for something outside of language.php
-			if ($class != '\\Joomla\\CMS\\Language\\Language' && $class != 'JText')
+			if ($class != '\\Joomla\\CMS\\Language\\Language' && $class != 'Text')
```
Comment (SharkyKZ) [inline libraries/src/Language/Language.php]:
```suggestion
			if ($class != '\\Joomla\\CMS\\Language\\Language' && $class != Text::class)
```

**Comment by SharkyKZ** [comment]
There's still many issues left. Maybe it would be simpler to do this by directory or extension rather than by class.

**Comment by C-Lodder** [comment]
@SharkyKZ I've fixed the remaining errors I could find. Other than that, it just looks like CS

File: libraries/src/Factory.php [hunk 13]
```diff
@@ -796,9 +796,9 @@ protected static function createDocument()
 	 * @param   string   $ua           UA User agent to use
 	 * @param   boolean  $uamask       User agent masking (prefix Mozilla)
 	 *
-	 * @return  \JStream
+	 * @return  \Stream
```
Comment (SharkyKZ) [inline libraries/src/Factory.php]:
```suggestion
	 * @return  Stream
```

File: libraries/src/Factory.php [hunk 14]
```diff
@@ -796,9 +796,9 @@ protected static function createDocument()
...
 	 *
-	 * @see     \JStream
+	 * @see     \Stream
```
Comment (SharkyKZ) [inline libraries/src/Factory.php]:
```suggestion
	 * @see     Stream
```

File: libraries/src/MVC/Model/LeagcyModelLoaderTrait.php [hunk 15]
```diff
@@ -137,7 +140,7 @@ public static function getInstance($type, $prefix = '', $config = array())
 
 			if (!class_exists($modelClass))
 			{
-				\JLog::add(\JText::sprintf('JLIB_APPLICATION_ERROR_MODELCLASS_NOT_FOUND', $modelClass), \JLog::WARNING, 'jerror');
+				\JLog::add(Text::sprintf('JLIB_APPLICATION_ERROR_MODELCLASS_NOT_FOUND', $modelClass), \JLog::WARNING, 'jerror');
```
Comment (brianteeman) [inline libraries/src/MVC/Model/LeagcyModelLoaderTrait.php:143]: Looks like JLog is not being found and changed by the script?

Comment (SharkyKZ) [reply libraries/src/MVC/Model/LeagcyModelLoaderTrait.php:143]: This PR is limited to several classes. See PR summary.

File: plugins/editors-xtd/pagebreak/pagebreak.php [hunk 16]
```diff
@@ -11,7 +11,6 @@
 
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Object\CMSObject;
```
Comment (SharkyKZ) [inline plugins/editors-xtd/pagebreak/pagebreak.php:14]: Restore.

File: plugins/editors-xtd/module/module.php [hunk 17]
```diff
@@ -11,7 +11,6 @@
 
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Object\CMSObject;
```
Comment (SharkyKZ) [inline plugins/editors-xtd/module/module.php:14]: Restore.

File: plugins/editors-xtd/fields/fields.php [hunk 18]
```diff
@@ -12,7 +12,6 @@
 use Joomla\CMS\Component\ComponentHelper;
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Object\CMSObject;
```
Comment (SharkyKZ) [inline plugins/editors-xtd/fields/fields.php:15]: Restore.

File: plugins/editors-xtd/contact/contact.php [hunk 19]
```diff
@@ -11,7 +11,6 @@
 
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Object\CMSObject;
```
Comment (SharkyKZ) [inline plugins/editors-xtd/contact/contact.php:14]: Restore.

File: plugins/editors-xtd/article/article.php [hunk 20]
```diff
@@ -11,7 +11,6 @@
 
 use Joomla\CMS\Factory;
 use Joomla\CMS\Language\Text;
-use Joomla\CMS\Object\CMSObject;
```
Comment (SharkyKZ) [inline plugins/editors-xtd/article/article.php:14]: Restore.

File: plugins/content/fields/fields.php [hunk 21]
```diff
@@ -7,7 +7,7 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
-defined('_JEXEC') or die();
+defined('_JEXEC') or die;
```
Comment (SharkyKZ) [inline plugins/content/fields/fields.php:10]: Unrelated change.

**Comment by SharkyKZ** [comment]
For classes that are only used as part of a docblock you could remove import statements and use fully qualified names instead. Not sure which is the preferred way. @joomla/maintainers?

For classes only used in inline comments use fully qualified names and no import statement.

File: modules/mod_feed/Helper/FeedHelper.php [hunk 22]
```diff
@@ -26,7 +27,7 @@ class FeedHelper
 	 *
 	 * @param   \Joomla\Registry\Registry  $params  module parameters
 	 *
-	 * @return  JFeedReader|string
+	 * @return  FeedReader|string
```
Comment (SharkyKZ) [inline modules/mod_feed/Helper/FeedHelper.php:30]: Such class doesn't exist?

File: modules/mod_breadcrumbs/Helper/BreadcrumbsHelper.php [hunk 23]
```diff
@@ -13,6 +13,7 @@
 
 use Joomla\CMS\Application\CMSApplication;
 use Joomla\CMS\Factory;
+use Joomla\CMS\Filesystem\Path;
```
Comment (SharkyKZ) [inline modules/mod_breadcrumbs/Helper/BreadcrumbsHelper.php:16]: Remove.

File: libraries/src/MVC/View/CategoryFeedView.php [hunk 24]
```diff
@@ -11,6 +11,7 @@
 defined('JPATH_PLATFORM') or die;
 
 use Joomla\CMS\Factory;
+use Joomla\CMS\Feed\Feed;
```
Comment (SharkyKZ) [inline libraries/src/MVC/View/CategoryFeedView.php]:
```suggestion
use Joomla\CMS\Document\Feed\FeedItem;
```
And then sort by alpha.

File: libraries/src/Language/Text.php [hunk 25]
```diff
@@ -12,6 +12,7 @@
 
 use Joomla\CMS\Factory;
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Language\Text;
```
Comment (SharkyKZ) [inline libraries/src/Language/Text.php:15]: Remove.

File: modules/mod_breadcrumbs/Helper/BreadcrumbsHelper.php [hunk 26]
```diff
@@ -54,7 +55,7 @@ public static function getList(Registry $params, CMSApplication $app)
 
 		$count = count($items);
 
-		// Don't use $items here as it references JPathway properties directly
+		// Don't use $items here as it references Pathway properties directly
```
Comment (SharkyKZ) [inline modules/mod_breadcrumbs/Helper/BreadcrumbsHelper.php:58]: Maybe use full qualified name here. Not sure if necessary for this comment.

File: libraries/src/Language/Language.php [hunk 27]
```diff
@@ -11,6 +11,7 @@
 defined('JPATH_PLATFORM') or die;
 
 use Joomla\CMS\Factory;
+use Joomla\CMS\Language\Text;
```
Comment (SharkyKZ) [inline libraries/src/Language/Language.php:14]: Remove.

File: libraries/src/Image/Image.php [hunk 28]
```diff
@@ -11,6 +11,8 @@
 
 defined('JPATH_PLATFORM') or die;
 
+use Joomla\CMS\Image\Image;
```
Comment (SharkyKZ) [inline libraries/src/Image/Image.php:14]: Remove.

File: libraries/src/Image/Image.php [hunk 29]
```diff
@@ -11,6 +11,8 @@
...
+use Joomla\CMS\Image\ImageFilter;
```
Comment (SharkyKZ) [inline libraries/src/Image/Image.php:15]: Remove.

File: libraries/src/Filesystem/Folder.php [hunk 30]
```diff
@@ -671,7 +672,7 @@ public static function listFolderTree($path, $filter, $maxLevel = 3, $level = 0,
 
 		if ($level == 0)
 		{
-			$GLOBALS['_JFolder_folder_tree_index'] = 0;
+			$GLOBALS['_Folder_folder_tree_index'] = 0;
```
Comment (SharkyKZ) [inline libraries/src/Filesystem/Folder.php:675]: Revert.

File: libraries/src/Filesystem/Folder.php [hunk 31]
```diff
@@ -681,7 +682,7 @@ public static function listFolderTree($path, $filter, $maxLevel = 3, $level = 0,
 			// First path, index foldernames
 			foreach ($folders as $name)
 			{
-				$id = ++$GLOBALS['_JFolder_folder_tree_index'];
+				$id = ++$GLOBALS['_Folder_folder_tree_index'];
```
Comment (SharkyKZ) [inline libraries/src/Filesystem/Folder.php:685]: Revert.

File: libraries/src/Feed/FeedParser.php [hunk 32]
```diff
@@ -10,6 +10,7 @@
 
 defined('JPATH_PLATFORM') or die;
 
+use Joomla\CMS\Feed\Feed;
```
Comment (SharkyKZ) [inline libraries/src/Feed/FeedParser.php:13]: Remove.

File: libraries/src/Feed/FeedFactory.php [hunk 33]
```diff
@@ -10,6 +10,7 @@
 
 defined('JPATH_PLATFORM') or die;
 
+use Joomla\CMS\Feed\FeedParser;
```
Comment (SharkyKZ) [inline libraries/src/Feed/FeedFactory.php:13]: Remove.

File: libraries/src/Feed/FeedEntry.php [hunk 34]
```diff
@@ -11,6 +11,10 @@
 defined('JPATH_PLATFORM') or die;
 
 use Joomla\CMS\Date\Date;
+use Joomla\CMS\Feed\Feed;
+use Joomla\CMS\Feed\FeedPerson;
```
Comment (SharkyKZ) [inline libraries/src/Feed/FeedEntry.php:15]: Remove both.

File: libraries/src/Feed/Feed.php [hunk 35]
```diff
@@ -8,9 +8,13 @@
 
 namespace Joomla\CMS\Feed;
 
-defined('JPATH_PLATFORM') or die();
+defined('JPATH_PLATFORM') or die;
 
 use Joomla\CMS\Date\Date;
+use Joomla\CMS\Feed\FeedEntry;
+use Joomla\CMS\Feed\FeedPerson;
```
Comment (SharkyKZ) [inline libraries/src/Feed/Feed.php:15]: Remove both.

File: libraries/src/Feed/Feed.php [hunk 36]
```diff
@@ -8,9 +8,13 @@
 
 namespace Joomla\CMS\Feed;
 
-defined('JPATH_PLATFORM') or die();
+defined('JPATH_PLATFORM') or die;
```
Comment (SharkyKZ) [inline libraries/src/Feed/Feed.php:11]: Unrelated change.

File: libraries/src/Application/ConsoleApplication.php [hunk 37]
```diff
@@ -10,9 +10,11 @@
 
 defined('JPATH_PLATFORM') or die;
 
+use DispatcherAwareTrait;
```
Comment (SharkyKZ) [inline libraries/src/Application/ConsoleApplication.php:13]: Remove.

File: libraries/src/Application/ConsoleApplication.php [hunk 38]
```diff
@@ -10,9 +10,11 @@
...
 use Joomla\CMS\Console;
 use Joomla\CMS\Extension\ExtensionManagerTrait;
 use Joomla\CMS\Factory;
+use Joomla\CMS\Input\Cli;
```
Comment (SharkyKZ) [inline libraries/src/Application/ConsoleApplication.php:17]: Remove.

File: libraries/src/Application/CliApplication.php [hunk 39]
```diff
@@ -10,13 +10,15 @@
 
 defined('JPATH_PLATFORM') or die;
 
+use DispatcherAwareTrait;
```
Comment (SharkyKZ) [inline libraries/src/Application/CliApplication.php:13]: Remove.

**Comment by roland-d** [comment]
> For classes that are only used as part of a docblock you could remove import statements and use fully qualified names instead. Not sure which is the preferred way. @joomla/maintainers?

I wonder how that could happen? What is in your docblock is used in your function or not?

My idea is we use imports for everything but whatever the powers may decide :)

File: libraries/bootstrap.php [hunk 40]
```diff
@@ -62,7 +64,6 @@
 	JLog::addLogger(['logger' => 'messagequeue'], JLog::ALL, ['jerror']);
 }
 
-// Register the Crypto lib
```
Comment (SharkyKZ) [inline libraries/bootstrap.php:65]: Unrelated change.

File: libraries/bootstrap.php [hunk 41]
```diff
@@ -9,6 +9,8 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\Language\Text;
```
Comment (SharkyKZ) [inline libraries/bootstrap.php:12]: Remove.

**Comment by SharkyKZ** [comment]
@roland-d This happens a lot. E.g. in classes when a function argument is an object but it's not typehinted. So you have it in the docblock but not in the code.

File: administrator/modules/mod_feed/Helper/FeedHelper.php [hunk 42]
```diff
@@ -26,7 +27,7 @@ class FeedHelper
 	 *
 	 * @param   JRegisty  $params  The parameters object.
 	 *
-	 * @return  JFeedReader|string  Return a JFeedReader object or a string message if error.
+	 * @return  FeedReader|string  Return a FeedReader object or a string message if error.
```
Comment (SharkyKZ) [inline administrator/modules/mod_feed/Helper/FeedHelper.php:30]: Such class doesn't exist?

File: administrator/components/com_joomlaupdate/restore_finalisation.php [hunk 43]
```diff
@@ -14,7 +14,7 @@
 namespace
 {
 	// Require the restoration environment or fail cold. Prevents direct web access.
-	defined('_AKEEBA_RESTORATION') or die();
+	defined('_AKEEBA_RESTORATION') or die;
```
Comment (SharkyKZ) [inline administrator/components/com_joomlaupdate/restore_finalisation.php:17]: Unrelated change.

File: administrator/components/com_templates/Model/TemplateModel.php [hunk 44]
```diff
@@ -1524,11 +1525,11 @@ public function getImage()
 
 			if (file_exists(Path::clean($path . $fileName)))
 			{
-				$JImage = new \JImage(Path::clean($path . $fileName));
+				$Image = new Image(Path::clean($path . $fileName));
```
Comment (SharkyKZ) [inline administrator/components/com_templates/Model/TemplateModel.php:1528]: Revert changes to the variable name?

File: libraries/src/Feed/Feed.php (hunk 36 above)
Comment (C-Lodder) [reply libraries/src/Feed/Feed.php:11]: Just leave it as it is. I remove it to assist eith the regex

File: libraries/src/MVC/Model/LeagcyModelLoaderTrait.php (hunk 15 above)
Comment (C-Lodder) [reply libraries/src/MVC/Model/LeagcyModelLoaderTrait.php:143]: @brianteeman, it's not JLog itself, but a class inside the single quotes. Another regex improvement that's required

File: administrator/modules/mod_feed/Helper/FeedHelper.php (hunk 42 above)
Comment (C-Lodder) [reply administrator/modules/mod_feed/Helper/FeedHelper.php:30]: It does exist

Comment (SharkyKZ) [reply administrator/modules/mod_feed/Helper/FeedHelper.php:30]: I think it should be `Feed`. See https://github.com/joomla/joomla-cms/blob/ca51f7b68dd918cd8560179e91a22729cee7e156/libraries/src/Feed/FeedFactory.php#L35

Comment (C-Lodder) [reply administrator/modules/mod_feed/Helper/FeedHelper.php:30]: Oh, the `J` class is either incorrect or removed.

File: administrator/components/com_joomlaupdate/restore_finalisation.php (hunk 43 above)
Comment (C-Lodder) [reply administrator/components/com_joomlaupdate/restore_finalisation.php:17]: keeping this as stated above

File: plugins/content/fields/fields.php (hunk 21 above)
Comment (C-Lodder) [reply plugins/content/fields/fields.php:10]: keeping as stated above

**Comment by C-Lodder** [comment]
Ok it looks like the regex needs improving to make this automated. I'm going to close this for now and someone with more regex experience can take over.

@SharkyKZ Sorry for wasting your time, but I really appreciate the time you took to review it.
